content/movie-review-pt1.md

```diff
@@ -0,0 +1,464 @@
+# Create a Basic Program: Part 1
+
+# Lesson Objectives
+
+*By the end of this lesson, you will be able to:*
+
+- Use basic Rust language features for encapsulating code
+- Deserialize instruction data into Rust data types
+- Execute different program logic for different types of instructions
+- Explain the structure of a smart contract on Solana
+
+# TL;DR
+
+- Most programs support multiple discrete instructions - you decide when writing your program what these instructions are and what data must accompany them
+- Rust enums are often used to represent discrete program instructions
+- You can use the `borsh` crate and the `derive` attribute to provide Borsh deserialization and serialization functionality to Rust structs
+- Rust `match` expressions help create conditional code paths based on the provided instruction
+
+# Overview
+
+One of the most basic elements of a Solana program is the logic for handling instruction data. Most programs support multiple related functions and use differences in instruction data to determine which code path to execute. For example, instruction data may indicate that a client wishes to create a new piece of data or delete an existing piece of data.
+
+Since instruction data is provided to your program's entry point as a byte array, it's common to create a Rust data type to represent instructions in a more usable format. This lesson will walk through how to set up such a type, how to deserialize the instruction data into this format, and how to execute the proper code path based on the instruction passed into the program's entry point.
+
+## Rust basics
+
+Before we dive into the specifics of a basic Solana program, let's talk about the Rust basics we'll be using throughout this lesson.
+
+### Variables
+
+Variable assignment in Rust happens with the `let` keyword.
+
+```rust
+let age = 33;
+```
+
+Variables in Rust by default are immutable, meaning a variable's value cannot be changed once it has been set. In order to create a variable that we'd like to change at some point in the future, we must make use of the `mut` keyword, which stands for mutable. Defining a variable with this keyword means that the value stored in it can change.
+
+```rust
+// compiler will throw error
+let age = 33;
+age = 34;
+
+// this is allowed
+let mut mutable_age = 33;
+mutable_age = 34;
+```
```

Comment:
@jamesrp13 I think this is the only one, but is good markdown practice to separate code blocks w/a new line

Reply:
Yeah. Just for the authors - readers will see everything formatted and it won't matter, but it definitely makes it easier for us to work with/edit